src/Microsoft.Health.Api/Extensions/HeaderExtensions.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Text;
+using EnsureThat;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Health.Api.Extensions
+{
+    public static class HeaderExtensions
+    {
+        public static int GetTotalHeaderLength(this IHeaderDictionary headers, Encoding headerEncoding)
+        {
+            // Per https://en.wikipedia.org/wiki/List_of_HTTP_header_fields, each header will be of the form
+            // headerKey: headerValues, and terminated by an end-of-line character sequence. The list of headers
+            // will be terminated by another end-of-line character sequence.
+            EnsureArg.IsNotNull(headers, nameof(headers));
+            EnsureArg.IsNotNull(headerEncoding, nameof(headerEncoding));
+
+            int headerDelimiterByteCount = headerEncoding.GetByteCount(": ");
+            int headerEndOfLineCharactersByteCount = headerEncoding.GetByteCount("\r\n");
+            int headerLength = 0;
+            foreach (KeyValuePair<string, StringValues> header in headers)
+            {
+                headerLength += headerEncoding.GetByteCount(header.Key)
+                    + headerDelimiterByteCount
+                    + headerEncoding.GetByteCount(header.Value.ToString())
```

Comment:
`StringValues.ToString()` returns comma-separated values where the commas weren't actually present in the input. However, this may be redoing logic for parsing headers. So I suppose my question is: how are `StringValues` instances created for a given header?
- If it splits comma-separated values in the header (such that `A=1,2,3` is the same as `new StringValues( new List<string> {  "1", "2", "3" })`), then we're probably ok as `ToString` is undoing this logic. But this seems unlikely, as I think different headers are formatted differently, and I can't imagine each header has its own split logic.
- If it is aggregating duplicate headers (such that headers `A=1,2` and `A=3` is the same as `new StringValues( new List<string> {  "1,2", "3" })`), then we need to count the size of each entry in `StringValues` individually

Reply:
Ah good catch! I did a bit of research but couldn't find anything specific on how it works. Doing it the second way should work for all scenarios so I've updated it.